plugins/source_docs.go

```diff
@@ -14,20 +15,98 @@ import (
 //go:embed templates/*.go.tpl
 var templatesFS embed.FS
 
+type SourceDocsFormat string
+
+const (
+	SourceDocsFormatMarkdown = "markdown"
+	SourceDocsFormatJSON     = "json"
+)
+
+var SourceDocsFormats = []SourceDocsFormat{
+	SourceDocsFormatMarkdown,
+	SourceDocsFormatJSON,
+}
+
+func (s SourceDocsFormat) String() string {
+	return string(s)
+}
+
+func (s SourceDocsFormat) Validate() error {
+	for _, f := range SourceDocsFormats {
+		if s == f {
+			return nil
+		}
+	}
+	return fmt.Errorf("invalid format: %v", s.String())
+}
+
 // GenerateSourcePluginDocs creates table documentation for the source plugin based on its list of tables
-func (p *SourcePlugin) GenerateSourcePluginDocs(dir string) error {
+func (p *SourcePlugin) GenerateSourcePluginDocs(dir string, format SourceDocsFormat) error {
 	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
 		return err
 	}
 
 	SetDestinationManagedCqColumns(p.Tables())
 
+	switch format {
+	case SourceDocsFormatMarkdown:
+		return p.renderTablesAsMarkdown(dir)
+	case SourceDocsFormatJSON:
+		return p.renderTablesAsJSON(dir)
+	default:
+		return fmt.Errorf("unsupported format: %v", format.String())
+	}
+}
+
+type jsonTable struct {
+	Name        string       `json:"name"`
+	Description string       `json:"description"`
+	Columns     []jsonColumn `json:"columns"`
+	Relations   []jsonTable  `json:"relations"`
+}
+
+type jsonColumn struct {
+	Name string `json:"name"`
+	Type string `json:"type"`
```

Comment:
Maybe add `IsPK bool` or something to that effect?

Reply:
Ah yes, good call 👍